docs/recipes/complex-journal-entries.md

```diff
@@ -0,0 +1,61 @@
+# Complex (Multi-Debit, Multi-Credit) Journal Entries
+
+TigerBeetle is designed for maximum performance. In order to keep it lean, the database only supports simple transfers with a single debit and a single credit.
+
+However, you'll probably run into cases where you want transactions with multiple debits and/or credits. If so, this page is for you!
+
+## 1-to-N
+
+Transactions that involve multiple debits and a single credit OR a single debit and multiple credits are relatively straightforward.
+
+You can use multiple linked transfers as depicted below.
+
+### Single Debit, Multiple Credits
+
+This example debits a single account and credits multiple accounts. It uses the following accounts:
+
+- A _source account_ `A`, on the `USD` ledger.
+- Three _destination accounts_ `X`, `Y`, and `Z`, on the `USD` ledger.
+
+| Ledger | Debit Account | Credit Account | Amount | `flags.linked` |
+| -----: | ------------: | -------------: | -----: | -------------: |
+|    USD |           `A` |            `X` |  10000 |           true |
+|    USD |           `A` |            `Y` |     50 |           true |
+|    USD |           `A` |            `Z` |     10 |          false |
+
+### Multiple Debits, Single Credit
+
+This example debits multiple accounts and credits a single account. It uses the following accounts:
+
+- Three _source accounts_ `A`, `B`, and `C` on the `USD` ledger.
+- A _destination account_ `X` on the `USD` ledger.
+
+| Ledger | Debit Account | Credit Account | Amount | `flags.linked` |
+| -----: | ------------: | -------------: | -----: | -------------: |
+|    USD |           `A` |            `X` |  10000 |           true |
+|    USD |           `B` |            `X` |     50 |           true |
+|    USD |           `C` |            `X` |     10 |          false |
+
+## Many-to-Many Transactions
+
+Transactions with multiple debits and multiple credits are a bit more tricky.
+
+To simplify the math, we can use a temporary "control account" as an intermediary.
+
+In this example, we'll use the following accounts:
+
+- Three _source accounts_ `A` and `B` on the `USD` ledger.
+- Three _destination accounts_ `X`, `Y`, and `Z`, on the `USD` ledger.
+- A _compound entry control account_ `Control` on the `USD` ledger.
```

Comment:
Maybe mention that `Control` can have `flags.debits_must_not_exceed_credits` set? Though it isn't necessary, so maybe that is an unnecessary distraction.

Reply:
Happy to include it or leave it out. It seems like if this is something we're going to really recommend to people, it might be worth having some account flag like `net_balance_must_be_zero` or _maybe_ allowing `debits_must_not_exceed_credits` and `credits_must_not_exceed_debits` to be set at the same time (although that would technically allow situations where both the debits and credits are increased at the same time, which we wouldn't want in this case).